classes/publication/DAO.php

```diff
@@ -432,10 +431,10 @@ protected function saveControlledVocab(array $values, int $publicationId)
      */
     protected function deleteControlledVocab(int $publicationId)
     {
-        $this->submissionKeywordDao->insertKeywords([], $publicationId);
-        $this->submissionSubjectDao->insertSubjects([], $publicationId);
-        $this->submissionDisciplineDao->insertDisciplines([], $publicationId);
-        $this->submissionAgencyDao->insertAgencies([], $publicationId);
+        ControlledVocab::query()
+            ->withSymbolics(ControlledVocab::getDefinedVocabSymbolic())
```

Comment:
Mixing symbolic names and setting names?

Reply:
Not exactly, here we just need to delete all the controlled vocabs for the given publication but we can do it without taking into the consideration of symbolics as we only need need assoc type and assoc id . Updated .